soc/silabs/silabs_s2/xg22/Kconfig.soc

```diff
@@ -13,12 +13,52 @@ config SOC_SERIES_EFR32BG22
 	help
 	  Silicon Labs EFR32BG22 (Blue Gecko) Series MCU
 
+config SOC_SERIES_EFR32MG22
+	bool
+	select SOC_SILABS_XG22
+	help
+	  Silicon Labs EFR32MG22 (Mighty Gecko) Series MCU
+
 config SOC_PART_NUMBER_EFR32BG22C224F512IM40
 	bool
 	select SOC_SERIES_EFR32BG22
 
+config SOC_PART_NUMBER_EFR32MG22C224F512IM40
```

Comment:
See recent discussion in https://github.com/zephyrproject-rtos/zephyr/pull/93285#discussion_r2221382981 -- these options should not have the `PART_NUMBER` infix. Existing socs can keep their current option names until we do a treewide refactor, but new socs should just use `SOC_<soc_name>` directly.